src/gui/N2Go_Gui.php

```diff
@@ -86,12 +86,12 @@ public function adminOptions()
         }
 
         if ($_SERVER['REQUEST_METHOD'] == 'POST') {
-            $this->save_option('n2go_formUniqueCode', $_POST['formUniqueCode']);
+            $this->save_option('n2go_formUniqueCode', htmlspecialchars($_POST['formUniqueCode']));
 
-            $widgetStyleConfig = json_decode(stripcslashes($_POST['widgetStyleConfig']));
```

Comment:
I see this was not used before but shouldn't we keep this and use it to avoid sql injections?

Reply:
Not fully sure what you mean. You are correct about that it was not used, it was intended to check if `n2go_widgetStyleConfig` option key needs update. I agree that this line 93:
```
$this->save_option('n2go_widgetStyleConfig', $_POST['widgetStyleConfig']);
```
Is not the best, as we pass `$_POST` data directly to database, but I was not sure how to make it more safe. Note that in database we save this as plain JSON in a text field (`longtext`)

I think I could run `json_decode` and later `json_encode` to strip out stuff which should not be there. What do you think?

Reply:
ok I get now. then it can stay as now since the input is done from admin backend anyways.